### PR TITLE
[PREVIEW] Extending fe functional test framework

### PR DESCRIPTION
**Pull request by noronhaa**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-3058

### Change description ###
Extending front end functional test infrastructure

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: test/functional-tests/config/conf.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      removeExistingJsonReportFile: true,
+      removeOriginalJsonReportFile:true
+    }
+  }],
+
+  useAllAngular2AppRoots: true,
+
+  SELENIUM_PROMISE_MANAGER: false,
+
+  ignoreSynchronization: true
```
Comment (hemantt) [inline test/functional-tests/config/conf.js]: Also, can this config file be appropriately named and placed in test/config folder?

File: package.json [hunk 2]
```diff
@@ -21,10 +21,10 @@
     "test:a11y": "echo 'TODO: Accessibility tests'",
     "test:nsp": "nsp check",
     "test:cucumber": "protractor test/config/cucumber.conf.js",
-    "test:smoke": "./bin/runSmokeTests.sh",
-    "test:smokeTest": "protractor test/config/smoke.conf.js",
-    "test:functional": "echo 'TODO: Enable functional tests' || ./bin/runFunctionalTests.sh",
-    "test:functionalTests": "protractor test/config/functional.conf.js",
+    "test:smokeTest": "./bin/runSmokeTests.sh",
```
[the remark this answers is not in the document]
Comment (noronhaa) [reply package.json]: Yup, think I accidentally removed than in the rebase

File: test/functional-tests/pageObjects/loginPage.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
+BasePage = require('./basePage.js')
+let CCDBanner = require('./caseListPage.js')
+const selfUrlPath = '/logqqqqqin';
```
[the remark this answers is not in the document]
Comment (noronhaa) [reply test/functional-tests/pageObjects/loginPage.js]: it's from a previous commit - rebase error, will change

File: test/functional-tests/config/conf.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (noronhaa) [reply test/functional-tests/config/conf.js]: best to remove

Comment (hemantt) [reply test/functional-tests/config/conf.js]: 👍

**Pull request merged**